fahrschein/src/main/java/org/zalando/fahrschein/domain/Subscription.java

```diff
@@ -26,6 +28,13 @@ public Subscription(String id, String owningApplication, Set<String> eventTypes,
         this.authorization = authorization;
     }
 
+    //Ignoring this constructor from introspection-based serialization and deserialization functionality,so that It doesn't
+    //interfere with default-all-args constructor.
+    @JsonIgnore
+    public Subscription(String id, String owningApplication, Set<String> eventTypes) {
```

Comment:
Instead of having a new constructor, we can re-use the existing all-args constructor. But given the `public` nature of `Subscription.java` this would give the clients an option to construct their own subscription object without going through `SubscriptionBuilder.java`

Example:

```
Subscription subscription = new Subscription( "subscription-id",
                                                  "application-name",
                                                  Set.of("event-type"));

nakadiClient.stream(subscription)
                .withStreamParameters(new StreamParameters().withBatchLimit(10).withMaxUncommittedEvents(10))
                .listen(TestEvent.class, new TestListener());
```